cupy/core/core.pyx

```diff
@@ -708,6 +708,11 @@ cdef class ndarray:
         Returns:
             tuple of arrays: Indices of elements that are non-zero.
 
+        .. warning::
+
+            This function may synchronize the device.
+            Avoid using it in a performance-critical code.
```

Comment:
I feel we don't want to suggest to avoid using these functions without an alternative solution. How about removing this phrase and just saying the fact that these functions synchronize the device?

Reply:
Done in f22fe440bd66cefdc9d941685eaed54cafc2e77b.